core/string/ustring.h

```diff
@@ -39,6 +39,89 @@
 #include "core/typedefs.h"
 #include "core/variant/array.h"
 
+/*************************************************************************/
+/*  Utility Functions                                                    */
+/*************************************************************************/
+
+// Not defined by std.
+// strlen equivalent function for char16_t * arguments.
+constexpr size_t strlen(const char16_t *p_str) {
+	const char16_t *ptr = p_str;
+	while (*ptr != 0) {
+		++ptr;
+	}
+	return ptr - p_str;
+}
+
+// strlen equivalent function for char32_t * arguments.
+constexpr size_t strlen(const char32_t *p_str) {
+	const char32_t *ptr = p_str;
+	while (*ptr != 0) {
+		++ptr;
+	}
+	return ptr - p_str;
+}
+
+// strlen equivalent function for wchar_t * arguments; depends on the platform.
+constexpr size_t strlen(const wchar_t *str) {
+	// Use static_cast twice because reinterpret_cast is not allowed in constexpr
+#ifdef WINDOWS_ENABLED
+	// wchar_t is 16-bit
+	return strlen(static_cast<const char16_t *>(static_cast<const void *>(str)));
+#else
+	// wchar_t is 32-bit
+	return strlen(static_cast<const char32_t *>(static_cast<const void *>(str)));
+#endif
+}
+
+constexpr size_t _strlen_clipped(const char *p_str, int p_clip_to_len) {
+	if (p_clip_to_len < 0) {
+		return strlen(p_str);
+	}
+
+	int len = 0;
+	while (len < p_clip_to_len && *(p_str++) != 0) {
+		len++;
+	}
+	return len;
+}
+
+constexpr size_t _strlen_clipped(const char32_t *p_str, int p_clip_to_len) {
+	if (p_clip_to_len < 0) {
+		return strlen(p_str);
+	}
+
+	int len = 0;
+	while (len < p_clip_to_len && *(p_str++) != 0) {
+		len++;
+	}
+	return len;
+}
+
+/*************************************************************************/
+/*  StrRange                                                             */
+/*************************************************************************/
+
+template <typename Element>
+struct StrRange {
+	const Element *c_str;
+	size_t len;
```

Comment:
I would have used a more explicit uint64 type. No need to use uint32 as it will be padded anyway. So it will be clearer that every size is supported.

Reply:
You mean use `uint64` instead of `usize`? That would be fairly unusual when dealing with arrays.

Reply:
Again, stylistic. But it removes ambiguity. Unusual where ? CowData use uint64_t for its size, LocalVector uses uint32_t

Reply:
Interesting. I'm not sure why that is. Possibly because they're designed to be easily exposeable to GDScript? 
Still, I think `size_t` is cleaner; it's very common to use that for sizes across c++.

Reply:
I know that it is the default in C++ standard library. It's just, why be ambiguous on the size of a type when you don't have to. Size_t probably always is 64bits nowadays anyway, this is why this is a style choice. Any choice will do, just raising a point :).

Reply:
I'll keep it as `size_t` for now. But let's keep the discussion open for other people to chime in!

Reply:
Style changes in core are inherently a **very** low priority, so I'm not too surprised there's mixing and matching of different names/formats. A lot of the discrepancies could come down to how several templates originally used `int`/32-bit storage logic instead of 64-bit, but I can't say for certain. In any case, `size_t` is perfectly acceptable, and is arguably the best option for logic/metrics having to deal with… Well, size!